dataSourceImpl/src/main/java/com/panhuk/dataSourceimpl/DatastorePreferencesImpl.kt

```diff
@@ -40,12 +40,7 @@ class DatastorePreferencesImpl(private val dataStore: DataStore<Preferences>) :
 
   override fun isFirstTimeAppOpened(): Flow<Boolean> {
     return dataStore.data.map { preferences: Preferences ->
```

Comment:
return dataStore.data[FIRST_TIME] == null

Reply:
There is no way how we can do it. We are using flow to check this value so also we return the Flow<Boolean> value and it would be impossible if we use it directly and the last we can data variable doens't know anything about array structure
<img width="519" alt="Screenshot 2021-12-23 at 10 28 49" src="https://user-images.githubusercontent.com/37262034/147211605-9a8efc6d-40c6-4b48-832e-0f2d977c0dd9.png">